### PR TITLE
Mesh_3 : use PMP::Side_of_triangle_mesh

### LINKED ISSUE #2084: Using  Side_of_triangle_mesh in Mesh_3
Mesh_3 `Polyhedral_mesh_domain_3` should be using `Side_of_triangle_mesh` as an implementation detail.  `TriangleAccessor_3` concept should also be replaced by `FaceListGraph`.
If we want to avoid a dependency between packages, we could leave `Polyhedral_mesh_domain_3` as his, (deprecated it?) and introduced a new class working on models of `FaceListGraph`. This new class can be put in PMP.

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

This PR fixes issue #2084 by using `PMP::Side_of_triangle_mesh` and its public API from `Polyhedral_mesh_domain_3` in Mesh_3.

It also fixes some compilation issues for the use of Mesh_3 with `Surface_mesh` and `Graph_with_descriptor_with_graph`.

## Release Management

* Affected package : Mesh_3, Demo
* Issue solved : fix #2084

### CODE REVIEW

File: Mesh_3/include/CGAL/Polyhedral_mesh_domain_3.h [hunk 1]
```diff
@@ -29,17 +29,17 @@
...
-
 #include <CGAL/Mesh_3/global_parameters.h>
 #include <CGAL/Mesh_3/Robust_intersection_traits_3.h>
-#include <CGAL/Mesh_3/Triangle_accessor_primitive.h>
-#include <CGAL/Triangle_accessor_3.h>
-#include <CGAL/AABB_tree.h>
-#include <CGAL/AABB_traits.h>
+
+#include <CGAL/boost/graph/Graph_with_descriptor_with_graph.h>
+#include <CGAL/Surface_mesh.h>
```
Comment (sloriot) [inline Mesh_3/include/CGAL/Polyhedral_mesh_domain_3.h]: Why is this header needed? A forward declaration might be enough IMO.

Comment (janetournois) [inline Mesh_3/include/CGAL/Polyhedral_mesh_domain_3.h]: commit 8aa15d3 arrived in the meantime...

**Comment by sloriot** [comment]
Is there any issue using OneFaceGraphPerTree=Tag_false  in `AABB_face_graph_triangle_primitive` instead of `Graph_with_descriptor_with_graph` for Surface_mesh?

**Comment by sloriot** [comment]
I answer myself to my question: the input graph is needed after a query outside of the tree where the extra information is stored.

**Comment by janetournois** [comment]
Travis failed, I'm taking care of it

**Comment by lrineau** [comment]
Strange Travis build error:
```
[ 31%] Building CXX object CMakeFiles/grid_d.dir/grid_d.cpp.o
/home/travis/build/CGAL/cgal/Generator/examples/Generator/random_points_in_tetrahedral_mesh_3.cpp:17:15: error: 
      no type named 'Polyhedron_3' in namespace 'CGAL'
typedef CGAL::Polyhedron_3<K>                                     Polyhedron;
        ~~~~~~^
```
https://travis-ci.org/CGAL/cgal/jobs/247415839#L4450

Maybe a `#include` has been removed from a header of Mesh_3, and this `.cpp` do not include `<CGAL/Polyhedron_3.h>`.

**Comment by lrineau** [comment]
There is yet another build error in Travis:
https://travis-ci.org/CGAL/cgal/jobs/248266953#L4486

I suggest you try to compile all the `Generator` examples.

**Comment by sloriot** [comment]
[Testsuite line](https://cgal.geometryfactory.com/CGAL/testsuite/results-4.12-Ic-4.shtml#Mesh_3_Examples) fully red.

**Comment by janetournois** [comment]
I have rebased the branch, and the latest commits should fix the testsuite.

**Pull request merged**